test/bats/test.bats

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bats
+
+load helpers
+
+@test "gatekeeper-controller-manager running" {
+  run kubectl -n gatekeeper-system wait --for=condition=Ready --timeout=60s pod/gatekeeper-controller-manager-0
+  assert_success
+}
+
+@test "constrainttemplates crd is established" {
+  run kubectl -n gatekeeper-system wait --for condition=established --timeout=60s crd/constrainttemplates.templates.gatekeeper.sh
+  assert_success
+}
+
+@test "apply config" {
+  run kubectl apply -f demo/basic/sync.yaml
+  assert_success
+}
+
+@test "required labels test" {
+  run kubectl apply -f demo/basic/templates/k8srequiredlabels_template.yaml
+  assert_success
+
+  while [[ -z $(kubectl -n gatekeeper-system wait --for condition=established --timeout=60s crd/k8srequiredlabels.constraints.gatekeeper.sh) ]]; do echo "waiting for crd" && sleep 1; done
+
+  run kubectl apply -f demo/basic/constraints/all_ns_must_have_gatekeeper.yaml
```

Comment:
We should move test constraints into a safe directory. I wouldn't suspect modifying the contents of `demo` to break e2e tests